### PR TITLE
Simplify the batching process

### PR DESCRIPTION
**Pull request by AlexisBallo2**

### CODE REVIEW

**Review by alexnicholsamu — APPROVED** [review]
lgtm

File: BERT_run.py [hunk 1]
```diff
@@ -13,105 +13,106 @@
...
+        loss.backward()
+        optimizer.step()
+        batch_num += 1
+        print(f'Batch Number: {batch_num} | Loss: {loss_print:.4f}')
+        losses.append(loss_print.item())
+        # if early_stopping(loss_print.item()):
+        #     print("Early stopping at epoch:", i, "batch:", batch_num)
+        #     break
+
+        # index = random.randrange(len(testsentences) - 1)
```
Comment (liam-hp) [inline BERT_run.py:71]: can we delete all this commented code? if its not relevant we should remove it

File: batches.py [hunk 2]
```diff
@@ -30,46 +30,51 @@ def make_training_batch(start_index):
 
     for idx in range(start_index, end_index):
         tokens_a_index, tokens_b_index = idx, idx + 1
-        tokens_a, tokens_b = token_list[tokens_a_index], token_list[tokens_b_index]  # grab sentence pair
+        tokens_a = token_list[tokens_a_index]
```
Comment (liam-hp) [inline batches.py:33]: what is tokens_a and tokens_b? Can we put some comments in here explaining it... its a bit of a headache looking at without context

Comment (liam-hp) [reply batches.py:33]: is this the NSP section?

File: preprocess.py [hunk 3]
```diff
@@ -7,7 +7,8 @@
 
 # new max length of our sentences
 # before was 512
-maxLength = 32
+# space for cls and sep
+maxLength = 32 - 2
```
Comment (liam-hp) [inline preprocess.py:11]: lowkey we should put all of the hyperparameters in hyperParam.py so that we can easily see and tweak as needed- not urgent tho ofc

File: trainBert.sbatch [hunk 4]
```diff
@@ -21,7 +19,8 @@ echo "Job ID: ${SLURM_JOB_ID}"
 echo "Node: ${SLURMD_NODENAME}" echo "Starting: "`date +"%D %T"` 
 echo "$USER"
 # Running your code here 
-python -Wignore BERT_run.py
```
Comment (liam-hp) [inline trainBert.sbatch:24]: lmao fair

**Review by liam-hp — COMMENTED** [review]
I still don't have a great sense of the batching from a broader perspective but we can chat ab that- if it still runs and works then LGTM

File: batches.py (hunk 2 above)
Comment (liam-hp) [reply batches.py:33]: i.e. are we doing masked word prediction as well as NSP?
